Mathlib/LinearAlgebra/Transvection.lean

```diff
@@ -70,6 +79,39 @@ theorem of_right_eq_zero (f : Module.Dual R V) :
   ext
   simp [transvection]
 
+theorem eq_id_of_finrank_le_one
+    [Module.Free R V] [Module.Finite R V] [StrongRankCondition R]
+    {f : Module.Dual R V} {v : V}
+    (hfv : f v = 0) (h1 : Module.finrank R V ≤ 1) :
+    transvection f v = LinearMap.id := by
+  rcases subsingleton_or_nontrivial R with hR | hR
+  · have : Subsingleton V := Module.subsingleton R V
+    subsingleton
```

Comment:
I don't think you use this:
```suggestion
```
(it would be available via [nontrivial_of_invariantBasisNumber](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/InvariantBasisNumber.html#nontrivial_of_invariantBasisNumber) if you did btw)